### DISCUSSION TITLE
How do you threaten the monthly version as a major update for YYYY.MM.patch versioning?

### DISCUSSION THREAD

**Discussion by Skaronator** [renovatebot/renovate#23235] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I use home assistant container tag with Renovate. The Home Assistant versioning looks like this: 2023.7.1. Where its year followed by the current month and lastly the patch for that month. Each Month gets a new release, which should be treated as a major update.

https://github.com/home-assistant/core/releases

Currently, renovate things this is SemVer and handle the year as major and the month as minor.

### What you tried so far.

I tried the following regex but it doesn't seem to work because "2023.7" is not a valid "number"?

```
 "packageRules": [
    {
      "matchPackageNames": ["ghcr.io/home-assistant/home-assistant"],
      "versioning": "regex:^(?<major>\\d+\\.\\d+)\\.(?<patch>\\d+)$"
    },
]
```

At least renovate want to downgrade from 2023.7.1 to 2023.1.7. Also `"newMajor": 2023` is parsed as a number not a string.

```
{
        "deps": [
          {
            "autoReplaceStringTemplate": "{{newValue}}{{#if newDigest}}@{{newDigest}}{{/if}}",
            "currentDigest": "sha256:a86ff5d05ce46520c53d67c8da55aba310de9b9b4ca8eead1ae0b5ab1c068f97",
            "currentValue": "2023.7.1",
            "currentVersion": "2023.7.1",
            "datasource": "docker",
            "depName": "ghcr.io/home-assistant/home-assistant",
            "fixedVersion": "2023.7.1",
            "isSingleVersion": true,
            "packageName": "ghcr.io/home-assistant/home-assistant",
            "registryUrl": "https://ghcr.io",
            "replaceString": "2023.7.1@sha256:a86ff5d05ce46520c53d67c8da55aba310de9b9b4ca8eead1ae0b5ab1c068f97",
            "sourceUrl": "https://github.com/home-assistant/core",
            "versioning": "regex:^(?<major>\\d+\\.\\d+)\\.(?<patch>\\d+)$",
            "warnings": [],
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "2023.1.7",
                "newValue": "2023.1.7",
                "newMajor": 2023,
                "newMinor": 0,
                "updateType": "patch",
                "newDigest": "sha256:56af74dddf922c16ce95ca3ca3f5638bdb173bc79d05d566aa68659f61ed5c16",
                "branchName": "renovate/ghcr.io-home-assistant-home-assistant-2023.x"
              }
            ]
          }
        ],
        "packageFile": "smart-home/home-assistant/values.yaml"
      },
```

---

Removing the dot (.) between year and month would probably cause issue once the month are double digit. e.g. 202312 > 20241

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
I think you should treat the years a major.minor.patch, even if minor releases can be breaking. You can hopefully then have packageRules to group it together with other breaking/major releases if you desire

**Reply by Skaronator** [reply, marked as answer]
Yeah probably fair enough. My packageRules looks like this. That should cover it.


```
  "packageRules": [
    {
      "matchUpdateTypes": ["digest", "pin", "pinDigest"],
      "automerge": true,
      "automergeType": "branch"
    },
    {
      "matchUpdateTypes": ["minor", "patch"],
      "automerge": true
    },
    {
      "matchUpdateTypes": ["major"],
      "minimumReleaseAge": "1 day"
    },
    {
      "matchPackageNames": [
        "ghcr.io/home-assistant/home-assistant",
        "ghcr.io/esphome/esphome"
      ],
      "matchUpdateTypes": ["major","minor"],
      "minimumReleaseAge": "1 day"
    }
  ],
```

**Reply by Skaronator** [reply]
It's still not working correctly. I didn't have time to check why, but it looks like matchUpdateTypes and minimumReleaseAge cannot be used at the same time: https://github.com/renovatebot/renovate/discussions/23270
